rs/execution_environment/src/canister_manager/tests.rs

```diff
@@ -6648,3 +6651,427 @@ fn memory_usage_updates_increment_subnet_available_memory() {
+
+    // Create a canister on each of the two subnets.
+    let canister_id1 = env1
+        .install_canister_with_cycles(
+            UNIVERSAL_CANISTER_WASM.to_vec(),
+            vec![],
+            None,
+            INITIAL_CYCLES_BALANCE,
+        )
+        .unwrap();
+    let canister_id2 = env2
+        .install_canister_with_cycles(
+            UNIVERSAL_CANISTER_WASM.to_vec(),
+            vec![],
+            Some(
+                CanisterSettingsArgsBuilder::new()
+                    .with_controllers(vec![PrincipalId::new_anonymous(), canister_id1.into()])
+                    .build(),
+            ),
+            INITIAL_CYCLES_BALANCE,
+        )
+        .unwrap();
+
+    (canister_id1, canister_id2)
+}
+
+/// Trigger a rename for a setup according to `install_two_universal_canisters`.
+fn rename_canister(
+    env1: &StateMachine,
+    env2: &StateMachine,
+    sender_canister: CanisterId,
+    old_canister_id: CanisterId,
+    new_canister_id: CanisterId,
+    sender_canister_version: u64,
+    new_version: u64,
+    new_num_changes: u64,
+    send_to_subnet: bool,
+) -> WasmResult {
+    const MAX_TICKS: usize = 100;
+    let user_id = user_test_id(1).get();
+
+    let arguments = RenameCanisterArgs {
+        canister_id: old_canister_id.into(),
+        rename_to: RenameToArgs {
+            canister_id: new_canister_id.into(),
+            version: new_version,
+            total_num_changes: new_num_changes,
+        },
+        sender_canister_version,
+    };
+
+    // Sending the request to the subnet should always work, sending to IC_00 works only if the
+    // routing table is correct.
+    let management_canister = if send_to_subnet {
+        CanisterId::from(env2.get_subnet_id())
+    } else {
+        IC_00
+    };
+
+    let msg_id = env1
+        .submit_ingress_as(
+            user_id,
+            sender_canister,
+            "update",
+            wasm()
+                .call_simple(
+                    management_canister,
+                    Method::RenameCanister,
+                    call_args()
+                        .other_side(arguments.encode())
+                        .on_reject(PayloadBuilder::default().reject_message().reject().build()),
+                )
+                .build(),
+        )
+        .unwrap();
+
+    env1.execute_round();
+    env2.execute_round();
+    env1.execute_round();
+    env2.execute_round();
+    env1.execute_round();
+
+    env1.await_ingress(msg_id, MAX_TICKS).unwrap()
+}
+
+#[test]
+fn can_rename_canister() {
+    const MAX_TICKS: usize = 100;
+    let user_id = user_test_id(1).get();
+
+    let (env1, env2) = two_subnets_simple();
+
+    // Create a canister on each of the two subnets.
+    let (canister_id1, canister_id2) = install_two_universal_canisters(&env1, &env2);
+
+    let test_blob: Vec<u8> = vec![42, 41];
+
+    // Modify the memory of the canister
+    env2.submit_ingress_as(
```

Comment:
```suggestion
    env2.execute_ingress_as(
```
I guess we want to execute the message here.